spring-client-common/pom.xml

```diff
@@ -12,8 +12,14 @@
 
   <dependencies>
     <dependency>
-      <groupId>io.camunda.spring</groupId>
-      <artifactId>spring-client-annotations</artifactId>
+      <groupId>org.apache.httpcomponents.client5</groupId>
+      <artifactId>httpclient5</artifactId>
+      <version>5.2.1</version>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.13.1</version>
     </dependency>
   </dependencies>
```

Comment:
Looking at the dependencies we could offer our clients independently from spring:
`camunda-sdk-java` and re-use them in a `camunda-sdk-spring` offering. 
This would unblock users from using the clients with other frameworks like quarkus, ktor etc.

Reply:
I like the idea, I'll check how to best structure the project modules to offer the `camunda-sdk-java`. Although the entire repo will be evolving first to Spring first SDK. 

I can probably see what can be done so that the main repo will be about Spring SDK, and then we can have a tiny or special module that can cater to non-spring apps but not explicitly (kinda similar on how we embed `Operate` in Spring-Zeebe and can offer it as a bean but we are not openly advertising it).

Reply:
Yes, that would be great. I can imagine to just have a json+http dependency in a java sdk for example.

Reply:
Ok I caved in @sbuettner  😁  Addressed in https://github.com/camunda-community-hub/spring-zeebe/pull/510/commits/f5e117b341f1297526050f79f8cbbf25ce19e66f. 

And I think we are on the right track.

So essentially it will now look like this

<img width="447" alt="Screenshot 2023-11-22 at 4 06 05 PM" src="https://github.com/camunda-community-hub/spring-zeebe/assets/133918079/56d882a4-5cfe-40a8-8119-11d1e8aaa817">

We now have a plain Java module to support clients other than Spring. But first priority will be given for Spring related tasks. 

I noticed that there is no sense to maintain the `spring-client-operate` module so removed it for now (since the `c8-java-sdk/java-client-operate`) and the `starter-camunda` handles the auto-configuration. If there is a need for custom annotations for `Operate` which I don't foresee at the moment, I can bring it back.

Reply:
+100! And just to double down: We should avoid "c8" as a name anyway - it is Camunda 8 or simply Camunda :-)